engine/src/main/java/org/terasology/math/MatrixUtils.java

```diff
@@ -102,159 +74,43 @@ public static FloatBuffer matrixToFloatBuffer(Matrix3f m, FloatBuffer fb) {
      * @param fb the float buffer to copy the matrix into
      * @return The provided float buffer.
      */
-    public static FloatBuffer matrixToFloatBuffer(Matrix4f m, FloatBuffer fb) {
-        fb.put(m.m00);
-        fb.put(m.m10);
-        fb.put(m.m20);
-        fb.put(m.m30);
-        fb.put(m.m01);
-        fb.put(m.m11);
-        fb.put(m.m21);
-        fb.put(m.m31);
-        fb.put(m.m02);
-        fb.put(m.m12);
-        fb.put(m.m22);
-        fb.put(m.m32);
-        fb.put(m.m03);
-        fb.put(m.m13);
-        fb.put(m.m23);
-        fb.put(m.m33);
-
-        fb.flip();
-        return fb;
+    public static FloatBuffer matrixToFloatBuffer(Matrix4fc m, FloatBuffer fb) {
+        return m.get(fb);
     }
 
-    public static org.joml.Matrix4f createViewMatrix(float eyeX, float eyeY, float eyeZ, float centerX, float centerY, float centerZ, float upX, float upY, float upZ) {
-        return createViewMatrix(new org.joml.Vector3f(eyeX, eyeY, eyeZ), new org.joml.Vector3f(centerX, centerY, centerZ), new org.joml.Vector3f(upX, upY, upZ));
+    public static Matrix4f createViewMatrix(
+            float eyeX, float eyeY, float eyeZ,
+            float centerX, float centerY, float centerZ,
+            float upX, float upY, float upZ
+    ) {
+        return new Matrix4f().setLookAt(
+                eyeX, eyeY, eyeZ,
+                centerX, centerY, centerZ,
+                upX, upY, upZ
+        );
     }
```

Comment:
NOTE:
The original implementation of this method did:
Matrix4f().setLookAt(...).setTranslation(-eyeX, -eyeY, -eyeZ)
I removed setting the translation component because I trust joml more. Opinions?

Reply:
I would probably try to get rid of MatrixUtils. It doesn't really add much especially if we are just wrapping a single function.

Reply:
`calcViewProjectionMatrix()` might still be useful to make these calculations less error prone. But I technically agree.